code/modules/mob/living/carbon/human/human_damage.dm

```diff
@@ -427,3 +427,17 @@ This function restores all organs.
 	apply_effect(10, EYE_BLUR)
 	apply_effect(10, WEAKEN)
 	update_canmove()
+
+/mob/living/carbon/human/apply_radiation(var/rads, var/application = EXTERNAL)
+	if(application == EXTERNAL)
+		var/contents = get_contents_in_object(src)
+
+		var/obj/item/device/geiger_counter/G = locate(/obj/item/device/geiger_counter) in contents
```

Comment:
You should give mobs an on_irradiated event, and have the geiger counter `Add` itself to it when it gets equipped (using the `equipped` proc) and remove itself when it gets unequipped.